src/main/java/se/sundsvall/casedata/service/scheduler/emailreader/EmailReaderWorker.java

```diff
@@ -47,34 +51,44 @@ public EmailReaderWorker(final MessageRepository repository, final ErrandReposit
 		this.notificationService = notificationService;
 	}
 
-	void getAndProcessEmails() {
+	List<Email> getAndProcessEmails() {
 
 		try {
-			emailReaderClient.getEmail(emailReaderProperties.municipalityId(), emailReaderProperties.namespace())
-				.forEach(this::saveAndRemoteDelete);
+			return emailReaderClient.getEmail(emailReaderProperties.municipalityId(), emailReaderProperties.namespace());
+
 		} catch (final Exception e) {
 			LOG.error("Error when fetching emails from EmailReader", e);
 		}
+		return emptyList();
 	}
 
-	@Transactional
-	public void saveAndRemoteDelete(final Email email) {
+	@Transactional(propagation = Propagation.REQUIRES_NEW)
+	public boolean save(final Email email) {
 		try {
 			final var errandNumber = parseSubject(email.getSubject());
 
 			errandRepository.findByErrandNumber(errandNumber)
 				.filter(errand -> !messageRepository.existsById(email.getId()))
 				.ifPresent(errand -> {
+					Hibernate.initialize(errand.getStakeholders());
```

Comment:
Is it possible to fetch stakeholders eagerly instead of using this? 
(i.e. modify the fetch mechanism in the entity class?)

Reply:
It's possible and certainly would be a nicer looking solution possibly. All though I'm uncertain if it's wanted behaviour to eagerly loading in stakeholders all the time, could potentially have some effects on performance and memory?

Reply:
I'm not sure? But in my opinion the stakeholder object does not contain any memory intensive structures?

Reply:
I suppose. I guess if we start experiencing issues with eagerly fetching we can rollback to this solution. Will change